version: '2'

services:
  peer-base:
    image: hyperledger/fabric-peer
    environment:
      - CORE_VM_ENDPOINT=unix:///host/var/run/docker.sock
      # the following setting starts chaincode containers on the same
      # bridge network as the peers
      # https://docs.docker.com/compose/networking/
      # ---CHANGED---


      #- CORE_LOGGING_LEVEL=ERROR
      - CORE_LOGGING_LEVEL=DEBUG
      - CORE_PEER_TLS_ENABLED=true
      # ERROR: context deadline execeeded - container gets deleted - container logs
      # https://jira.hyperledger.org/browse/FAB-6937
      - CORE_VM_DOCKER_HOSTCONFIG_NETWORKMODE=acmedevmode_basic
      - CORE_PEER_GOSSIP_USELEADERELECTION=true
      - CORE_PEER_GOSSIP_ORGLEADER=false
      - CORE_PEER_PROFILE_ENABLED=true
      # https://github.com/hyperledger/caliper/issues/143
      - CORE_CHAINCODE_STARTUPTIMEOUT=6000000000000000000000000000s
      - CORE_CHAINCODE_EXECUTETIMEOUT=6000000000000000000000000000s
      - CORE_PEER_TLS_CERT_FILE=/etc/hyperledger/fabric/tls/server.crt
      - CORE_PEER_TLS_KEY_FILE=/etc/hyperledger/fabric/tls/server.key
      - CORE_PEER_TLS_ROOTCERT_FILE=/etc/hyperledger/fabric/tls/ca.crt
      - CORE_VM_DOCKER_ATTACHSTDOUT=true
      # ERROR: Not true info or debug - noticed from the docker peer logs
      - CORE_CHAINCODE_LOGGING_LEVEL=DEBUG
      - CORE_CHAINCODE_LOGGING_SHIM=DEBUG
    working_dir: /opt/gopath/src/github.com/hyperledger/fabric/peer
    command: peer node start
